sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -203,13 +214,15 @@ def list_configuration_settings(self, *args, **kwargs) -> ItemPaged[Configuratio
         if select:
             select = ["locked" if x == "read_only" else x for x in select]
         snapshot_name = kwargs.pop("snapshot_name", None)
+        tags = kwargs.pop("tags_filter", None)
 
         try:
             if snapshot_name is not None:
-                return self._impl.get_key_values(  # type: ignore
+                return self._impl.get_key_values(  # type: ignore[return-value]
```

Comment:
Why do we need to type ignore this?

Reply:
I would suspect it's because we're passing in `cls` to change the deserialized return type, and so this method return type no longer matches the generated method return type.

Reply:
Then our typing is wrong?